### PR TITLE
Running node in debug mode while testing

### PR DESCRIPTION
**Pull request by krasimir**
Hello everyone!

**Problem**
~~I wanted to make an addition to one of the packages. Everything was cool but I wasn't able to run the test in watching mode (or at least I didn't find that bit in the docs)~~
I wanted to run node in a debug mode so I can use the debugger while running the tests.

**Solution**
~~Mocha by default has a `--watch` option which is really nice because we don't have to create a gulp task for monitoring files. What I did is to introduce a new environment variable called `TEST_WATCH` which allow us to run~~
I added TEST_DEBUG to be recognized in `scripts/test.sh`.
```
TEST_ONLY=<package> TEST_DEBUG=true make test
```
It simply appends `debug` to the Mocha's call in `scripts/test.sh`.

**UAT**
1. Checkout my branch
2. Run `TEST_ONLY=babel-cli TEST_DEBUG=true make test-only`
3. You'll see that the tests are not proceeding and node stops inside the script. Press `c` and hit Enter to get them executed.

**Notes**
One of the modules contained a symbolic links to paths like `/Users/sebmck/Projects/babel/regenerator/node_modules/mocha/mocha.css`. I amended those so they don't point to an absolute location. That by the way breaks the watching mode of Mocha.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -66,6 +66,12 @@ This is mostly overkill and you can limit the package to a select by using the `
 $ TEST_ONLY=babel-cli make test
 ```
 
+If want to run tests in continues mode:
```
Comment (hzoo) [inline CONTRIBUTING.md]: Maybe `If you want to run tests continuously`

Comment (krasimir) [inline CONTRIBUTING.md]: :+1: [done](https://github.com/krasimir/babel/commit/eb269537d73ca5c208ac91cc69ad1b38cd2daac6)

**Comment by hzoo** [comment]
Although with this you still have to run `make watch` to compile the files, and then `TEST_ONLY=babel-cli TEST_WATCH=true make test`

**Comment by krasimir** [comment]
@hzoo actually you are right. I'm indeed running `make watch` in a separate terminal which makes `TEST_WATCH` pointless. However, I just made another commit to the same branch. It uses the same code changes but introduces TEST_DEBUG. It's about running node in a debug mode which is a must-have feature for me. 

I'll amend the title and description of the PR too.

**Comment by hzoo** [comment]
I could see the benefit of a `test watch` if it would auto run the tests on save (instead of running `make test` each time) but it either compile the files as well or you would still have to run `make watch`

**Comment by krasimir** [comment]
Yep, that's right and most of the tests are based on actual and expected file assertion. I personally don't change those files very often. So TEST_WATCH is indeed not needed.

**Comment by krasimir** [comment]
What you think about the TEST_DEBUG option.

**Comment by kittens** [comment]
@babel/core: Have any of you needed anything like this? Probably not much point if this is extremely niche when you can just patch it into debug mode yourself.

**Comment by krasimir** [comment]
I noticed that I'm constantly running the tests in debug and in normal mode. Switching between the two is possible with hitting :arrow_up: once or twice in the terminal (running `make test-only` with and without `TEST_DEBUG`). I agree that I can patch that by myself but if there is no `TEST_DEBUG` then I'll have to amend `scripts/test.sh` every time. 

Anyways guys, feel free to close the PR if that's not needed. I can always remove that bit from my code while proceeding with my next PR.

**Comment by hzoo** [comment]
ok goina close for now - if more people think they need this we can reopen

**Comment by krasimir** [comment]
:+1:

**Pull request closed without merging**